cg/services/run_devices/abstract_classes.py

```diff
@@ -83,6 +87,27 @@ def post_process(self, run_name: str, dry_run: bool = False):
         """Store sequencing metrics in StatusDB and relevant files in Housekeeper."""
         pass
 
+    def post_process_all(self, run_names: list[str], dry_run: bool = False) -> bool:
+        """Post-process all PacBio runs in the sequencing directory."""
+        LOG.info("Starting post-processing for all runs")
+        is_post_processing_successful = True
+        for run_name in run_names:
+            if self._is_run_post_processed(run_name):
+                LOG.info(f"Run {run_name} has already been post-processed")
+                continue
```

Comment:
move this into the self.post_process

Reply:
I receive conflicting arguments now, as @islean said we agreed to put this in the post_process_all. I think there is merit in keeping it here

Reply:
ah I see it is on the ABC... I don't think this is the right approach more and more. We just seem to need to find work arounds and copy logic into places.

the self._is_run_post_processed should also happen in the self.post_post_process.

Getting more and more in favour of fetching the services in the CLI and just going through the dirs and call self.post_process.

Reply:
To clarify - I do not want this in the post_process_all - I want this functionality in `RunNamesService.get_run_names`. I was also not envisioning the implementation of `post_process_all` to be put on the parent `PostProcessingService`, as I do agree that this is a limiting design choice. But on the other hand, I can not see what steps in the parts of `post_process_all` which would differ between the devices if the functionality of the post_process_all would be (excluding error handling)

```
runs = self.run_names_service.get_run_names
for run in runs:
    self.post_process(run)
```

Reply:
`I want this functionality in `RunNamesService.get_run_names` the run names should just return the list of directories and should not care about its state of post processing. Otherwise you require extra functionality to fetch directories for cleaning that does not have this check in place for example.

Reply:
Good point - then I want another method in the service called `get_runs_to_post_process` or something which does care about the state of post processing. 

I do think that the functionality ensuring that all previously post processed runs are not post processed again should be happening either in
1. picking up runs, or
2. ensuring that post processing can start.

I am pretty certain that it was suggested to be added in 2. but it was rebuked. What is the benefit of adding it in the actual post_process command?